src/type_set.rs

```diff
@@ -33,13 +30,15 @@ macro_rules! type_set {
 #[derive(Default)]
 pub struct TypeSet<T> {
     // Stored in left to right traversal order of the type tree.
-    names: Vec<&'static str>,
+    pub names: Vec<&'static str>,
```

Comment:
Why was this made public? `TypeSet<T>` is meant to be largely opaque to the user, so I think it should be kept private. If we really need it to be accessible outside of this module, then you can make it `pub(crate)` instead 🙂

Reply:
Good point, i fix this.